packages/react-menu/src/components/Menu/useMenu.tsx

```diff
@@ -20,10 +29,15 @@ const mergeProps = makeMergeProps<MenuState>({ deepMerge: menuShorthandProps });
  * {@docCategory Menu }
  */
 export const useMenu = (props: MenuProps, ref: React.Ref<HTMLElement>, defaultProps?: MenuProps): MenuState => {
+  const { document } = useFluent();
+  const triggerId = useId();
```

Comment:
_These are related a bit: https://github.com/microsoft/fluentui/pull/17387/files#r595403952_

---

We can have [`useTriggerElement`](https://github.com/microsoft/fluentui/blob/4503e2d3f62f785363b5937bfb5ce278c27ab0fe/packages/fluentui/react-bindings/src/hooks/useTriggerElement.tsx#L32) hook as in Northstar that will return us `ReactElement` for sure.

In this case, we can just use:

```suggestion
  const triggerElement = useTriggerElement();
  const triggerId = useId('popup-', triggerElement.props.id);
```

Anyway, it will be awesome to have test that ensures that we are handling `id`s passed by users.